libs/http-ext/src/lib/http-ext.ts

```diff
@@ -90,24 +95,19 @@ export class HttpExt {
   /**
    * Throw an exception if the plugin condition return type is invalid.
    */
-  private _throwIfInvalidCondition({
+  private _throwIfInvalidPluginCondition({
     plugin,
-    request,
-    index
+    request
   }: {
     plugin: Plugin;
     request: HttpExtRequest;
-    index: number;
   }): void {
     if (
       isFunction(plugin.condition) &&
       isBoolean(plugin.condition({ request })) === false
     ) {
```

Comment:
Not sure if we should do this.
At must we could check that `condition` is `null`, `undefined` or a `function`.
For the moment, it is really surprising to see that the condition is called multiple times for the same request. Once here (and when building the error) and once when really running the condition.
The other thing is that the condition can return a promise or observable.

What do you think?

Reply:
I refactored a bit this part, now condition is evaluated once and it also handle asynchronous computations.

Reply:
perfect!